pkg/v1/tkg/tkgpackageclient/repository_update.go

```diff
@@ -14,10 +14,31 @@ import (
 	kappipkg "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/packaging/v1alpha1"
 	versions "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/versions/v1alpha1"
 
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/log"
 	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/tkgpackagedatamodel"
 )
 
 func (p *pkgClient) UpdateRepository(o *tkgpackagedatamodel.RepositoryOptions, progress *tkgpackagedatamodel.PackageProgress, operationType tkgpackagedatamodel.OperationType) {
+	p.updateRepository(o, progress, operationType)
+}
+
+func (p *pkgClient) UpdateRepositorySync(o *tkgpackagedatamodel.RepositoryOptions, operationType tkgpackagedatamodel.OperationType) error {
+	pp := getDefaultProcessConfig()
+
+	go p.updateRepository(o, pp, operationType)
+
+	initialMsg := fmt.Sprintf("Updating package repository '%s'", o.RepositoryName)
+	if err := DisplayProgress(initialMsg, pp); err != nil {
+		if err.Error() == tkgpackagedatamodel.ErrRepoNotExists {
+			log.Warningf("package repository '%s' does not exist in namespace '%s'. Consider using the --create flag to add the package repository", o.RepositoryName, o.Namespace)
```

Comment:
As your ```*sync``` functions don't get called from the cmd, you probably don't the following log message?
```
Consider using the --create flag to add the package repository
```